realm/realm-library/src/androidTest/java/io/realm/RealmResultsTests.java

```diff
@@ -1027,4 +1029,459 @@ public void deleteAndDeleteAll() {
+        });
+    }
+
+    @Test
+    public void size_asEmptyListAfterClassRemoved() {
+        final String CLASS_NAME = "KingsAndQueens";
+        DynamicRealm realm = initializeDynamicRealm(CLASS_NAME);
+
+        // make sure we have only one object
+        RealmResults<DynamicRealmObject> results = realm.where(CLASS_NAME).findAll();
+        assertEquals(results.size(), 1);
+
+        // now the result should be empty
+        removeClassFromDynamicRealm(realm, CLASS_NAME);
+        assertEquals(0, results.size());
+        assertTrue(results.isEmpty());
+        realm.close();
+    }
+
+    @Test
+    public void size_asEmptyListAfterClassRemovedInOtherThread() {
+        final String CLASS_NAME = "KingsAndQueens";
+        final CountDownLatch bgRealmClosed = new CountDownLatch(1);
+        DynamicRealm realm = initializeDynamicRealm(CLASS_NAME);
+
+        // make sure we have only one object
+        RealmResults<DynamicRealmObject> results = realm.where(CLASS_NAME).findAll();
+        assertEquals(results.size(), 1);
+
+        // remove class in background
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                DynamicRealm bgRealm = DynamicRealm.getInstance(realmConfig);
+                removeClassFromDynamicRealm(bgRealm, CLASS_NAME);
+                bgRealm.close();
+                bgRealmClosed.countDown();
+            }
+        }).start();
+
+        TestHelper.awaitOrFail(bgRealmClosed);
+        assertEquals(0, results.size());
+        assertTrue(results.isEmpty());
+        realm.close();
+    }
+
+    @Test
+    public void contains_asEmptyListAfterClassRemoved() {
+        final String CLASS_NAME = "KingsAndQueens";
+        DynamicRealm realm = initializeDynamicRealm(CLASS_NAME);
+
+        // make sure we contains an object in RealmResults
+        RealmResults<DynamicRealmObject> results = realm.where(CLASS_NAME).findAll();
+        DynamicRealmObject object = realm.where(CLASS_NAME).findFirst();
+        assertTrue(results.contains(object));
+
+        // now the result should be empty
+        removeClassFromDynamicRealm(realm, CLASS_NAME);
+        assertFalse(results.contains(object));
+        realm.close();
+    }
+
+    @Test
+    public void contains_asEmptyListAfterClassRemovedInOtherThread() {
+        final String CLASS_NAME = "KingsAndQueens";
+        final CountDownLatch bgRealmClosed = new CountDownLatch(1);
+        DynamicRealm realm = initializeDynamicRealm(CLASS_NAME);
+
+        // make sure we contains an object in RealmResults
+        RealmResults<DynamicRealmObject> results = realm.where(CLASS_NAME).findAll();
+        DynamicRealmObject object = realm.where(CLASS_NAME).findFirst();
+        assertTrue(results.contains(object));
+
+        // remove class in background
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                DynamicRealm bgRealm = DynamicRealm.getInstance(realmConfig);
```

Comment:
I would use `realm.getConfiguration()` instead. Right now it isn't clear that these two Realms have the same config.